packages/core/src/storages/request_manager_tandem.ts

```diff
@@ -0,0 +1,162 @@
+import type { Dictionary } from '@crawlee/types';
+
+import type { Log } from '@apify/log';
+
+import { log } from '../log';
+import type { Request, Source } from '../request';
+import type { IRequestList } from './request_list';
+import type {
+    AddRequestsBatchedOptions,
+    AddRequestsBatchedResult,
+    IRequestManager,
+    RequestQueueOperationInfo,
+    RequestQueueOperationOptions,
+    RequestsLike,
+} from './request_provider';
+
+/**
+ * A request manager that combines a RequestList and a RequestQueue.
+ * It first reads requests from the RequestList and then, when needed,
+ * transfers them in batches to the RequestQueue.
+ */
+export class RequestManagerTandem implements IRequestManager {
+    private log: Log;
+    private requestList: IRequestList;
+    private requestQueue: IRequestManager;
+
+    constructor(requestList: IRequestList, requestQueue: IRequestManager) {
+        this.log = log.child({ prefix: 'RequestManagerTandem' });
+        this.requestList = requestList;
+        this.requestQueue = requestQueue;
+    }
+
+    /**
+     * Transfers a batch of requests from the RequestList to the RequestQueue.
+     * Handles both successful transfers and failures appropriately.
+     * @private
+     */
+    private async transferNextBatchToQueue(): Promise<void> {
```

Comment:
The `batch` is currently always of length 1, which doesn't sound too efficient.

Reply:
Yeah, I chose to do this to preserve the current behavior - I want to go one step at a time. This is just a private method and we can always make the batches bigger if we decide that it's a good idea.